client/src/components/History/model/syncCurrentHistoryToGalaxy.js

```diff
@@ -1,28 +1,28 @@
 // Sync Galaxy store to legacy galaxy current history
 
-import { getGalaxyInstance } from "app";
-import { switchMap, pluck } from "rxjs/operators";
-import { monitorChange } from "utils/observable/monitorChange";
-import { monitorBackboneModel } from "utils/observable/monitorBackboneModel";
+import { fromEvent } from "rxjs";
+import { map, filter, switchMap, pluck, distinctUntilChanged } from "rxjs/operators";
 
-// prettier-ignore
-export function syncCurrentHistoryToGalaxy(handler) {
-
-    // wait for the current history panel to appear
-    const currentHistoryPanel$ = monitorChange(() => {
-        return getGalaxyInstance()?.currHistoryPanel?.model;
-    });
-
-    // then emit the id each time it changes
-    const result$ = currentHistoryPanel$.pipe(
-        switchMap(model => monitorBackboneModel(model, "id").pipe(
-            pluck("id")
-        ))
+export function syncCurrentHistoryToGalaxy(galaxy$, store) {
+    // prettier-ignore
+    const historyId$ = galaxy$.pipe(
+        pluck("currHistoryPanel"),
+        switchMap((panel) => {
+            // relationship between currHistoryPanel and its model is murky, but I don't care, I'm
+            // just going to grab the id fresh after every possible event
+            return fromEvent(panel, "all").pipe(
```

Comment:
Awesome, I didn't know what event to monitor,  `"all"` makes sense. This is working great now.